### ISSUE TITLE
iojs 3.3.0 and Buffer

### ISSUE THREAD

**Issue by rahulpowar** [patriksimek/vm2#12]
Heads up - testing vm2 with the latest iojs has flagged an odd issue with Buffer, probably due to the various V8 changes related to Buffer.

````
var VM = require('vm2').NodeVM;

var buf1 = new Buffer(1);
console.log('check before VM:', (buf1 instanceof Buffer));

var vm = new VM();
vm.run("console.log('ran VM')");

var buf2 = new Buffer(1);
console.log('check after VM:', (buf2 instanceof Buffer), (buf1 instanceof Buffer));
````
The output of this is...

Node 0.12.7
````
check before VM: true
ran VM
check after VM: true true
````

iojs 3.3.0
````
check before VM: true
ran VM
check after VM: false true
````

`Buffer` is no longer instanceof-able and plays havoc with future API calls e.g. fs.readSync due to this check https://github.com/nodejs/node/blob/8f58fb92fff904a6ca58fd0df9ee5a1816e5b84e/lib/fs.js#L589

With Node4 moving forward from an iojs base, this is probably worth looking at.

**Comment by patriksimek** [comment]
With Node 4 I get:

```
check before VM: true
ran VM
check after VM: true true
```

But there are different issues with Node 4 I have to look into.

**Comment by rahulpowar** [comment]
Yeah, looks like the `instanceof` check works but things are not quite right with `Buffer` yet.

```
const VM = require('vm2').NodeVM;

const buf1 = new Buffer(1);
const bufPre = new Buffer('c3RpbGwgYnJva2Vu', 'base64');
console.log('bufPre out:', bufPre.toString('utf8'), (buf1 instanceof Buffer));

const vm = new VM();
vm.run("console.log('ran VM')");

const buf2 = new Buffer(1);
console.log('check after VM:', (buf2 instanceof Buffer), (buf1 instanceof Buffer));

const bufPost = new Buffer('c3RpbGwgYnJva2Vu', 'base64');
console.log('bufPost out:', bufPost.toString('utf8'));
```

Node 0.12.7

```
bufPre out: still broken true
ran VM
check after VM: true true
bufPost out: still broken
```

Node v4.1.0

```
bufPre out: still broken true
ran VM
check after VM: true true
buffer.js:536
        return this.base64Write(string, offset, length);
                    ^

TypeError: argument should be a Buffer
    at TypeError (native)
    at Buffer.write (buffer.js:536:21)
    at fromString (buffer.js:106:26)
    at new Buffer (buffer.js:53:12)
    at Object.<anonymous> (/Users/rahul/Downloads/vm2t/vm2-test.js:13:17)
    at Module._compile (module.js:434:26)
    at Object.Module._extensions..js (module.js:452:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Function.Module.runMain (module.js:475:10)
```

Native call seems to fail the type assertion.

**Comment by patriksimek** [comment]
Should be fixed in 0.2.4.
